pkg/server/apiextensions.go

```diff
@@ -187,106 +282,118 @@ func (c *inheritanceCRDLister) GetWithContext(ctx context.Context, name string)
 		}
 
 		if crd == nil {
-			return nil, apierrors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, "")
+			return nil, apierrors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, name)
 		}
 
 		return crd, nil
 	}
 
-	crdKey := clusters.ToClusterAwareKey(cluster.Name, name)
-	crd, err = c.crdLister.Get(crdKey)
-	if err != nil && !apierrors.IsNotFound(err) {
-		// something went wrong w/the lister - could only happen if meta.Accessor() fails on an item in the store.
-		return nil, err
+	org, workspace, err := helper.ParseLogicalClusterName(cluster.Name)
+	if err != nil {
+		return nil, fmt.Errorf("error determining workspace name from cluster name %q: %w", cluster.Name, err)
 	}
 
-	// If we found the CRD in ctx's logical cluster, that takes priority.
-	if crd != nil {
-		return crd, nil
-	}
+	systemCRDKeys := c.systemCRDProvider.Keys(ctx, org, workspace)
 
-	// ClusterWorkspace CRD is apparently not installed
-	if c.workspaceLister == nil {
-		return nil, apierrors.NewNotFound(apiextensionsv1.Resource("customresourcedefinitions"), name)
+	systemCRDKeyName := clusters.ToClusterAwareKey(SystemCRDLogicalCluster, name)
+	if systemCRDKeys.Has(systemCRDKeyName) {
+		crd, err = c.crdLister.Get(systemCRDKeyName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			// something went wrong w/the lister - could only happen if meta.Accessor() fails on an item in the store.
+			return nil, err
+		}
+
+		if crd != nil {
+			return crd, nil
+		}
+
+		return nil, apierrors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, name)
 	}
 
-	if !helper.HasClusterWorkspace(org, ws) {
-		// Without a cluster workspace no inheritance is possible
-		return nil, apierrors.NewNotFound(apiextensionsv1.Resource("customresourcedefinitions"), name)
+	// Priority 1: see if it comes from any APIBindings
```

Comment:
Maybe this is priority #2 (with system crds being #1), in keeping with the semantics for `List`?

Reply:
done